app/src/main/kotlin/com/wire/android/ui/login/Login.kt

```diff
@@ -0,0 +1,204 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.systemuicontroller.rememberSystemUiController
+import com.wire.android.R
+import com.wire.android.ui.common.AnimatedButtonColors
+import com.wire.android.ui.common.textfield.WireTextField
+
+
+@Preview
+@Composable
+fun LoginScreen() {
+    LoginContent()
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun LoginContent() {
+    val systemUiController = rememberSystemUiController()
+    val useDarkIcons = MaterialTheme.colors.isLight
+    val backgroundColor = MaterialTheme.colors.background
+
+    var email by remember { mutableStateOf(TextFieldValue("")) }
+    var password by remember { mutableStateOf(TextFieldValue("")) }
+
+    SideEffect { systemUiController.setSystemBarsColor(color = backgroundColor, darkIcons = useDarkIcons) }
+
+    Scaffold(topBar = { LoginTopBar() }) {
+        Column(modifier = Modifier.padding(16.dp)) {
+            Column(
+                modifier = Modifier.weight(1f, true),
+                verticalArrangement = Arrangement.Center,
+            ) {
+                EmailInput(modifier = Modifier.fillMaxWidth(), email = email, onEmailChange = { email = it })
+                Spacer(modifier = Modifier.height(16.dp))
+                PasswordInput(modifier = Modifier.fillMaxWidth(), password = password, onPasswordChange = { password = it })
+                Spacer(modifier = Modifier.height(16.dp))
+                ForgotPasswordLabel(modifier = Modifier.fillMaxWidth())
+            }
+            LoginButton(modifier = Modifier.fillMaxWidth(), email = email.text, password = password.text)
+        }
+    }
+}
+
+@Composable
+private fun EmailInput(modifier: Modifier, email: TextFieldValue, onEmailChange: (TextFieldValue) -> Unit) {
+    WireTextField(
+        value = email,
+        onValueChange = onEmailChange,
+        placeholderText = stringResource(R.string.login_email_placeholder),
+        labelText = stringResource(R.string.login_email_label),
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Text,
+            imeAction = ImeAction.Next
+        ),
+        modifier = modifier,
+    )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun PasswordInput(modifier: Modifier, password: TextFieldValue, onPasswordChange: (TextFieldValue) -> Unit) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+    var passwordVisibility by remember { mutableStateOf(false) }
```

Comment:
all these three are equivalent:
```
val mutableState = remember { mutableStateOf(default) }
var value by remember { mutableStateOf(default) }
val (value, setValue) = remember { mutableStateOf(default) }
```

I think that the one I used is the clearest one but I'm open to change if someone thinks otherwise 😉

Reply:
I am just not familiar with remember function I think, but when I read I think that remember { mutableStateOf(default) } return a object that is mutable itself, and if you add var infront of it , to me it kind of makes me think you want to mutate object that is mutable in itself, I dont know maybe just me

Reply:
I do not think you should declare somewhere in the code something like this :

value = someNewMutableStateOf(), that is why I think making it var instead of val misses the point ?

Reply:
That's why you can make `val item = remember{...}` and you can also use `var item by remember {...}` - `by` uses getter (`return this.value`) and setter (`this.value = value`) under the hood and thanks to that you can simplify your code by just using `item = "abc"` instead of `item.value = "abc"`

version with `by` is actually used in the documentation so this is Google's suggested way 😉 
You can read more here: https://developer.android.com/jetpack/compose/state#state-in-composables

Reply:
ahhhhh I see :P I did not see the by keyword, idk to me item.value makes more sense and is more readable